README.md

```diff
@@ -277,15 +278,45 @@ will be interpeted or passed along.
 All single word arguments (`-j4`, `--impure` etc) 
 are passed to the underlying nix invocation.
 
-##### Tracked files
+#### Tracked files
+
+`nix-direnv` makes a performance tradeoff 
+and only considers changes in a limited number of files 
+when deciding to update its cache.
+
+- for `use nix` this is:
+    * `~/.direnvrc`
+    * `~/.config/direnv/direnvrc`
+    * `.envrc`, 
+    * A single nix file. In order of preference:
+        + The file argument to `use nix`
+        + `default.nix` if it exists
+        + `shell.nix` if it exists
+
+- for `use flake` this is:
+    * `~/.direnvrc`
+    * `~/.config/direnv/direnvrc`
+    * `.envrc`
+    * `flake.nix`
+    * `flake.lock`
+    * `devshell.toml` if it exists
+
+To add more files to be checked use `nix_direnv_watch_file` like this
+
+```shell
+$ nix_direnv_watch_file your-file.nix
+```
 
-`nix-direnv` makes a performance tradeoff only considers changes in a limited
-number of files when deciding to update its cache.
+Or - if you don't mind the overhead (runtime and conceptual) of watching all nix-files:
+
+```shell
+$ nix_direnv_watch_file $(find . -name "*.nix")
```

Comment:
This might break if some directory contains spaces. Maybe xargs is better here?

```suggestion
$ nix_direnv_watch_file $(find . -name "*.nix")
```

Reply:
A very valid point. Thank you for bringing it up.
xargs does not necessarily re-use the calling shell, so `nix_direnv_watch_file` won't always exist.

I think we can do `find . -name "*.nix" -printf '"%p\n"'` though?

Reply:
Ended up going with the `-printf` solution. Find it in [71535f7](https://github.com/nix-community/nix-direnv/pull/223/commits/71535f7c357fd704fe67355f84be947e2371984e)